simulator/short-circuit/src/main/java/com/powsybl/incubator/simulator/shortcircuit/ShortCircuitNormIec.java

```diff
@@ -84,8 +97,234 @@ public double getKtT3W(ThreeWindingsTransformer t3w, int numLeg) {
+            throw new PowsyblException("Generator '" + gen.getId() + "' has wrong type to be adjusted");
+        }
+        double ikQmax = extensions2.getIkQmax();
+        double maxR1ToX1Ratio = extensions2.getMaxR1ToX1Ratio();
+        double cq = extensions2.getCq();
+
+        // Zq = cq * Unomq / (sqrt3 * Ikq)
+        // Zq = sqrt(r² + x²) which gives x = Zq / sqrt((R/X)² + 1)
+        double zq = cq * gen.getTerminal().getVoltageLevel().getNominalV() / (Math.sqrt(3.) * ikQmax / 1000.); // ikQmax is changed from A to kA
+        double xq = zq / Math.sqrt(maxR1ToX1Ratio * maxR1ToX1Ratio + 1.);
+        double rq = xq * maxR1ToX1Ratio;
+
+        extension.setDirectTransX(xq);
+        extension.setDirectSubtransX(xq);
+        extensions2.setTransRd(rq);
+        extensions2.setSubTransRd(rq);
+    }
+
+    public void generateZofLoadfromAsynch(Load load) {
+        LoadShortCircuit extension = load.getExtension(LoadShortCircuit.class);
+        if (extension == null) {
+            throw new PowsyblException("Load '" + load.getId() + "' could generate Z for short circuit because of missing extension input data");
+        }
+
+        double ratedMechanicalPower = extension.getRatedMechanicalP();
+        double ratedPowerFactor = extension.getRatedPowerFactor(); // cosPhi
+        double ratedS = extension.getRatedS();
+        double ratedU = extension.getRatedU();
+        double efficiency = extension.getEfficiency() / 100.; // conversion from percentages
+        double iaIrRatio = extension.getIaIrRatio();
+        double rxLockedRotorRatio = extension.getRxLockedRotorRatio();
+        int polePairNumber = extension.getPolePairNumber();
+
+        // Zn = 1/(Ilr/Irm) * Urm / (sqrt3 * Irm) = 1/(Ilr/Irm) * Urm² / (Prm / (efficiency * cosPhi))
+        // Xn = Zn / sqrt(1+ (Rm/Xm)²)
+        double zn = 1. / iaIrRatio * ratedU * ratedU / (ratedMechanicalPower / (efficiency * ratedPowerFactor));
+        double xn = zn / Math.sqrt(rxLockedRotorRatio * rxLockedRotorRatio + 1.);
+        double rn = xn * rxLockedRotorRatio;
+
+        /*System.out.println(" ================>  LoadId = " + load.getId() + " zn = " + zn + " xn = " + xn + " rn = " + rn);*/
+
+        // zn is transformed into a load that will give the equivalent zn in the admittance matrix
+        //using formula P(MW) = Re(Z) * |V|² / |Z|² and Q(MVA) = Im(Z) * |V|² / |Z|²
+        // TODO: once load will not be aggregated in the lfNetwork,
+        //  the info regarding the load with Asynchronous machine info should remain carried as Zn to fill the admittance matrix
+        double uNom = load.getTerminal().getVoltageLevel().getNominalV();
+        double pEqScLoad = rn * uNom * uNom / (zn * zn);
+        double qEqScLoad = xn * uNom * uNom / (zn * zn);
+
+        /*System.out.println(" ================>  LoadId = " + load.getId() + " pEqScLoad = " + pEqScLoad + " qEqScLoad = " + qEqScLoad);*/
+
+        load.setQ0(qEqScLoad);
```

Comment:
@JB-H we should not do that, p0 and q0 are LF inputs that should not be overwritten for short circuit calculation

Reply:
Agreed, but we need to modify the aggregation of loads in lfNetwork and modify the converter from iidm to lfNetwork to handle properly extra data related to short circuits and loads.
This is true in general for all modifications we make through the application of the norm in the commit below.